lib/model-key-composer.js

```diff
@@ -0,0 +1,52 @@
+// Copyright IBM Corp. 2016. All Rights Reserved.
+// Node module: loopback-connector
+// This file is licensed under the MIT License.
+// License text available at https://opensource.org/licenses/MIT
+
+'use strict';
+
+var debug = require('debug')('loopback:connector:model-key-composer');
+
+/**
+ * Build a single key string from a tuple (modelName, key).
+ *
+ * This method is typically used by KeyValue connectors to build a single
+ * key string for a given modelName+key tuple.
+ *
+ * @param {String} modelName
+ * @param {String} key
+ * @returns {String}
+ */
+exports.compose = function composeKeyFromModelNameAndKey(modelName, key) {
+  // Escape model name to prevent collision
+  //  'model' + 'foo:bar' --vs-- 'model:foo' + 'bar'
```

Comment:
Ah, there should have been two spaces to indent the example by one level. Will fix.